bip-0340.mediawiki

```diff
@@ -200,7 +200,7 @@ Input:
 * The signatures ''sig<sub>1..u</sub>'': ''u'' 64-byte arrays
 
 The algorithm ''BatchVerify(pk<sub>1..u</sub>, m<sub>1..u</sub>, sig<sub>1..u</sub>)'' is defined as:
-* Generate ''u-1'' random integers ''a<sub>2...u</sub>'' in the range ''1...n-1''. They are generated deterministically using a [https://en.wikipedia.org/wiki/Cryptographically_secure_pseudorandom_number_generator CSPRNG] seeded by a cryptographic hash of all inputs of the algorithm, i.e. ''seed = seed_hash(pk<sub>1</sub>..pk<sub>u</sub> || m<sub>1</sub>..m<sub>u</sub> || sig<sub>1</sub>..sig<sub>u</sub> )''. A safe choice is to instantiate ''seed_hash'' with SHA256 and use [https://tools.ietf.org/html/rfc8439 ChaCha20] with key ''seed'' as a CSPRNG to generate 256-bit integers, skipping integers not in the range ''1...n-1''.
+* Generate ''u-1'' random integers ''a<sub>2...u</sub>'' in the range ''1...n-1''. They are generated deterministically using a [https://en.wikipedia.org/wiki/Cryptographically_secure_pseudorandom_number_generator CSPRNG] seeded by a cryptographic hash of all inputs of the algorithm, i.e. ''seed = seed_hash(pk<sub>1</sub>..pk<sub>u</sub> || m<sub>1</sub>..m<sub>u</sub> || sig<sub>1</sub>..sig<sub>u</sub> )''. A safe choice is to instantiate ''seed_hash'' with SHA256 and use [https://tools.ietf.org/html/rfc8439 ChaCha20] with key ''seed'' as a CSPRNG to generate 128-bit integers.<ref>Randomizers are not required to be larger than 128-bit to achieve the desired security level of 128 bits (see [[bip-0340/batch-randomizers.mediawiki|lemma]]).</ref>
```

Comment:
Updated sentences are good. This still says `Generate ''u-1'' random integers ''a<sub>2...u</sub>'' in the range ''1...n-1''` in the first sentence. I mean it's not wrong, not sure if you wanted to keep this on purpose but if yes, then the updated sentences are kind of unexpected to the reader.

I'm not sure what the best thing is. We could simply say `1...2^128-1`. That's simple and clean but not sure if this is too restrictive. What if this is harder in your implementation than just generating full scalars. 

Nit: In any case, let's change "random" -> "uniformly random"

Here's a suggestion to play with (if you agree with the idea to specify both options):
```Generate ''u-1'' uniformly random integers ''a<sub>2...u</sub>'' in the range ''1...n-1'', or alternatively in the range ''1...2^128-1''.<ref>Integers are not required to be larger than 128 bits to achieve the desired security level of 128 bits (see [[bip-0340/batch-randomizers.mediawiki|lemma]]). Shorter integers can speed up computations in optimized implementations.</ref>``` ...

Next sentence (not your PR but we could fix it here):
At the moment, we don't really say that they must be generated deterministically or not. Maybe:
`It is recommended to generate them deterministically `... ?

Another question (again not your PR, sorry :P ) : Do we allow `0...n-1` (or `0...2^128-1`)? 
Pro: It doesn't matter and it's much easier to implement. If I was given a BIP that says `1...`, I might just ignore it and implement `0...`. Also, if we write `0...`, then it's somewhat cleaner that using 2^128 bit integers work.
Con: If you use a real PRG, you may want to reject 0 because it's dangerous. But it's dangerous anyway to use a real PRG?

I lean towards `0...`.

Reply:
> not sure if you wanted to keep this on purpose

No, that was an oversight. Good ideas, updated commit with your suggestions.